appsec/src/helper/subscriber/waf.cpp

```diff
@@ -124,9 +124,111 @@ void log_cb(DDWAF_LOG_LEVEL level, const char *function, const char *file,
         std::string_view(message, message_len));
 }
 
-void extract_tags_and_metrics(parameter_view diagnostics, std::string &version,
-    std::map<std::string, std::string> &meta,
-    std::map<std::string_view, double> &metrics)
+std::string waf_update_init_report_tags(
+    bool success, std::optional<std::string> rules_version)
+{
+    std::string tags{"success"sv};
+    if (success) {
+        tags += ":true"sv;
+    } else {
+        tags += ":false"sv;
+    }
+    if (rules_version) {
+        tags += ",event_rules_version:"sv;
+        tags += rules_version.value();
+    }
+    tags += ",waf_version:"sv;
+    tags += ddwaf_get_version();
+    return tags;
```

Comment:
Yeah, this is what I meant before with leaking the telemetry API specifics, I think you could solve this by creating a small abstraction on top of the tags which then takes care of creating the final string.

Reply:
see above, but I don't really find it necessary to abstract away such a simple convention